demo/CMakeLists.txt

```diff
@@ -0,0 +1,29 @@
+set(demos
+        "allegro5"
+#        "common"
+#        "d3d11"
+#        "d3d12"
+#        "d3d9"
+#        "gdi"
```

Comment:
Possibly have these as individual options defaulting to if the platform dependencies are found?